developer-docs-site/docs/guides/nfts/mint-nft-cli.md

```diff
@@ -4,587 +4,291 @@ title: "Mint NFTs with the Aptos CLI"
-
-## 3. Add admin account and functions
-
-Publishing a module under a resource account means that we will not be able to update the module, and the module will be immutable and autonomous. We need administrator accounts, because we give up control over the smart contracts when we deploy a module using a resource account. Admin accounts and functions let you update the configuration going forward.
-
-If you will need to update the configuration of the module, we suggest adding a few admin functions to control the configurations of the module. You will also need to add the admin account's address to the module's `Move.toml` file so the admin functions can confirm the caller to the admin functions is actually a valid administrator.
-
-This work maps to the demonstration in [reate_nft_with_resource_and_admin_accounts.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/mint_nft/3-Adding-Admin/sources/create_nft_with_resource_and_admin_accounts.move).
-
-### Create admin account
-
-Here we create an admin account and include two admin functions so we can set and update when/if we want to enable token minting for this contract.
-
-Run `aptos-init` once again, this time to create an admin account:
-
-```shell
-`aptos init --profile admin`
+```rust
+struct MintConfiguration has key {
+    collection_name: String,
+    token_name: String,
+    token_uri: String,
+}
 ```
 
-### Publish the module under a resource account
+We give `MintConfiguration` the `key` ability so that the module contract can store it at an account's resources, which the module can find later if supplied with the account address. The account address we store it to will be the creator of the collection.
 
-1. In your `aptos-core` source checkout, navigate to the `aptos-core/aptos-move/move-examples/mint_nft/3-Adding-Admin` directory:
+Some collection configuration parameters will be hard-coded in the contract as `const` variables:
 
-```shell
-cd aptos-move/move-examples/mint_nft/3-Adding-Admin
+```rust
+    const COLLECTION_DESCRIPTION: vector<u8> = b"Your collection description here!";
+    const TOKEN_DESCRIPTION: vector<u8> = b"Your token description here!";
+    // ...other const values
+    const TOKENS_BURNABLE_BY_CREATOR: bool = false;
+    const TOKENS_FREEZABLE_BY_CREATOR: bool = false;
 ```
 
-2. Edit `Move.toml` in that directory to replace `admin_addr = "0xcafe"` with the new `admin` address from `.aptos/config.yaml`.
+You can make these configurable on your own by adding them to the list of parameters sent into `initialize_collection` function below.
 
-3. Run the following CLI command to publish the module under a resource account, updating the `<seed>` and `default-account-address>` with your own:
+### Writing a collection creation function
 
-```shell
-aptos move create-resource-account-and-publish-package --seed <seed> --address-name mint_nft --profile default --named-addresses source_addr=<default-account-address>
+```rust
```

Comment:
Let's not have this entire snippet here (any changes later would need to be made to both the code and this doc). Just the aptos_token::create_collection call should be good.